node/internal.ts

```diff
@@ -914,3 +917,64 @@ export function _normalizeSeparators(p: string): string {
     // remove redundant slashes
     return p.replace(/\/\/+/g, '/');
 }
+
+//-----------------------------------------------------
+// Expose proxy information to vsts-node-api
+//-----------------------------------------------------
+export function _exposeProxySettings(): void {
+    let proxyUrl: string = _getVariable('Agent.ProxyUrl');
+    if (proxyUrl && proxyUrl.length > 0) {
+        let proxyUsername: string = _getVariable('Agent.ProxyUsername');
+        let proxyPassword: string = _getVariable('Agent.ProxyPassword');
+        let proxyBypassHostsJson: string = _getVariable('Agent.ProxyBypassList');
+
+        global['_vsts_task_lib_proxy_url'] = proxyUrl;
+        global['_vsts_task_lib_proxy_username'] = proxyUsername;
+        global['_vsts_task_lib_proxy_bypass'] = proxyBypassHostsJson;
+        global['_vsts_task_lib_proxy_password'] = _exposeTaskLibSecret(proxyPassword);
+        global['_vsts_task_lib_proxy'] = true;
+    }
+}
+
+//-----------------------------------------------------
+// Expose certificate information to vsts-node-api
+//-----------------------------------------------------
+export function _exposeCertSettings(): void {
+    let ca: string = _getVariable('Agent.CAInfo');
+    if (ca) {
+        global['_vsts_task_lib_cert_ca'] = ca;
+    }
+
+    let clientCert: string = _getVariable('Agent.ClientCert');
+    if (clientCert) {
+        let clientCertKey: string = _getVariable('Agent.ClientCertKey');
+        let clientCertArchive: string = _getVariable('Agent.ClientCertArchive');
+        let clientCertPassword: string = _getVariable('Agent.ClientCertPassword');
+
+        global['_vsts_task_lib_cert_clientcert'] = clientCert;
+        global['_vsts_task_lib_cert_key'] = clientCertKey;
+        global['_vsts_task_lib_cert_archive'] = clientCertArchive;
+        global['_vsts_task_lib_cert_passphrase'] = _exposeTaskLibSecret(clientCertPassword);
+    }
+
+    if (ca || clientCert) {
+        global['_vsts_task_lib_cert'] = true;
+    }
+}
+
+// We store the secret on disk and hold the encryption key and storage file in memory
+// return base64encoded<filePath>:base64encoded<encryptionKey>
+// downstream vsts-node-api will retrieve the secret later
+function _exposeTaskLibSecret(secret: string): string {
+    if (secret) {
+        let encryptKey = crypto.randomBytes(256);
+        let cipher = crypto.createCipher("aes-256-ctr", encryptKey);
+        let encryptedContent = cipher.update(secret, "utf8", "hex");
+        encryptedContent += cipher.final("hex");
+
+        let storageFile = path.join(_getVariable('Agent.TempDirectory'), uuidV4());
```

Comment:
can we update the other .taskkey location to temp as well?
Fall back to OS temp.  But that means the file names can't have guids.